open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/model/PeriodicLimit.java

```diff
@@ -0,0 +1,173 @@
+            case FORTNIGHT:
+                return now.plus(Duration.ofDays(14));
+            case MONTH:
+                return now.plus(Period.ofMonths(1));
+            case HALF_YEAR:
+                return now.plus(Period.ofMonths(6));
+            case YEAR:
+                return now.plus(Period.ofYears(1));
+            default:
+                throw new IllegalArgumentException("Invalid PeriodType");
+        }
+    }
+
+    /**
+     * Calculates the expiry time for a calendar based on the period type.
+     *
+     * @param now the current time
+     * @return the expiry time for a calendar
+     */
+    private Instant calculateExpiryTimeForCalendar(Instant now) {
+        LocalDate localDate = now.atZone(ZoneId.systemDefault()).toLocalDate();
+        PeriodicTypesEnum periodType = PeriodicTypesEnum.valueOf(this.periodType.toUpperCase());
+        switch (periodType) {
+            case DAY:
+                return localDate.plusDays(1).atStartOfDay(ZoneId.systemDefault()).toInstant();
+            case WEEK:
+                return localDate.with(TemporalAdjusters.next(DayOfWeek.SUNDAY)).plusDays(1).
+                        atStartOfDay(ZoneId.systemDefault()).toInstant();
+            case FORTNIGHT:
+                return now.plus(Duration.ofDays(14));
+            case MONTH:
+                return localDate.with(TemporalAdjusters.firstDayOfNextMonth()).
+                        atStartOfDay(ZoneId.systemDefault()).toInstant();
+            case HALF_YEAR:
+                return calculateHalfYearExpiry(localDate);
+            case YEAR:
+                return localDate.with(TemporalAdjusters.firstDayOfNextYear()).atStartOfDay(ZoneId.systemDefault())
+                        .toInstant();
+            default:
+                throw new IllegalArgumentException("Invalid PeriodType");
+        }
+    }
+
+    /**
+     * Calculates the expiry time for a half year.
+     *
+     * @param localDate the current date
+     * @return the expiry time for a half year
+     */
+    private Instant calculateHalfYearExpiry(LocalDate localDate) {
+        Month currentMonth = localDate.getMonth();
+        if (currentMonth.getValue() < 7) {
+            return localDate.withMonth(6).with(TemporalAdjusters.lastDayOfMonth()).atStartOfDay(ZoneId.systemDefault())
+                    .toInstant();
+        } else {
+            return localDate.withMonth(12).with(TemporalAdjusters.lastDayOfMonth()).atStartOfDay(ZoneId.systemDefault())
+                    .toInstant();
+        }
+    }
+
+    /**
+     * Calculates and sets the cyclic paid amount based on the period alignment.
+     */
+    private void calculateCyclicPaidAmount() {
+        if (periodAlignment.equalsIgnoreCase(ConsentExtensionConstants.CONSENT)) {
+            cyclicRemainingAmount = BigDecimal.valueOf(0);
+        } else if (periodAlignment.equalsIgnoreCase(ConsentExtensionConstants.CALENDAR )) {
+            LocalDate now = LocalDate.now();
```

Comment:
instead of getting the current date everytime shall we assign once to a variable and use it